### PR TITLE
Move m_TeleOuts from ddr controller to game controller

### PR DESCRIPTION
**Pull request by ChillerDragon**
#7127

## Checklist

- [ ] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/game/server/gamecontroller.cpp [hunk 1]
```diff
@@ -38,6 +38,8 @@ IGameController::IGameController(class CGameContext *pGameServer)
 	m_ForceBalanced = false;
 
 	m_CurrentRecord = 0;
+
+	InitTeleporter();
```
Comment (heinrich5991) [inline src/game/server/gamecontroller.cpp:42]: By putting this here, you moved the place where this is executed. Have you tested that this still works, ingame on a map with teleporters?

**Review by heinrich5991 — COMMENTED** [review]
LGTM otherwise.

File: src/game/server/gamecontroller.cpp (hunk 1 above)
Comment (ChillerDragon) [reply src/game/server/gamecontroller.cpp:42]: No this is untested

Comment (ChillerDragon) [reply src/game/server/gamecontroller.cpp:42]: Tested the change in game. Tele still works. Also put a breakpoint into InitTeleporter to show the diff between this pr and master and it looks close enough. No unexpected detour:

# current master

```
gef➤  bt
#0  dbg_break () at /home/chiller/Desktop/git/ddnet/src/base/system.cpp:208
#1  0x0000555555606386 in CGameControllerDDRace::InitTeleporter (this=0x5555559f2ab0) at /home/chiller/Desktop/git/ddnet/src/game/server/gamemodes/DDRace.cpp:224
#2  0x0000555555604ffd in CGameControllerDDRace::CGameControllerDDRace (this=0x5555559f2ab0, pGameServer=0x7ffff477d010) at /home/chiller/Desktop/git/ddnet/src/game/server/gamemodes/DDRace.cpp:22
#3  0x00005555555f098a in CGameContext::OnInit (this=0x7ffff477d010, pPersistentData=0x0) at /home/chiller/Desktop/git/ddnet/src/game/server/gamecontext.cpp:3601
#4  0x00005555555a6ec3 in CServer::Run (this=0x7ffff57d5010) at /home/chiller/Desktop/git/ddnet/src/engine/server/server.cpp:2707
#5  0x00005555555927db in main (argc=0x1, argv=0x7fffffffd178) at /home/chiller/Desktop/git/ddnet/src/engine/server/main.cpp:197
```

# this pr

```
gef➤  bt
#0  dbg_break () at /home/chiller/Desktop/git/ddnet/src/base/system.cpp:208
#1  0x00005555556023c0 in IGameController::InitTeleporter (this=0x5555559f2ab0) at /home/chiller/Desktop/git/ddnet/src/game/server/gamecontroller.cpp:736
#2  0x00005555555ff369 in IGameController::IGameController (this=0x5555559f2ab0, pGameServer=0x7ffff477d010) at /home/chiller/Desktop/git/ddnet/src/game/server/gamecontroller.cpp:42
#3  0x00005555556051cd in CGameControllerDDRace::CGameControllerDDRace (this=0x5555559f2ab0, pGameServer=0x7ffff477d010) at /home/chiller/Desktop/git/ddnet/src/game/server/gamemodes/DDRace.cpp:18
#4  0x00005555555f09f2 in CGameContext::OnInit (this=0x7ffff477d010, pPersistentData=0x0) at /home/chiller/Desktop/git/ddnet/src/game/server/gamecontext.cpp:3601
#5  0x00005555555a6ef3 in CServer::Run (this=0x7ffff57d5010) at /home/chiller/Desktop/git/ddnet/src/engine/server/server.cpp:2707
#6  0x000055555559280b in main (argc=0x1, argv=0x7fffffffd178) at /home/chiller/Desktop/git/ddnet/src/engine/server/main.cpp:197

```

So the only code in between the old and the new tele are those variable definitions

https://github.com/ChillerDragon/ddnet/blob/a02525a4d097e9c46c9a2921085c569e875e8d71/src/game/server/gamecontroller.cpp#L23-L40

And the call to DoWarmup() which also just sets variables.

Comment (ChillerDragon) [reply src/game/server/gamecontroller.cpp:42]: I also force pushed just to trigger the CI again after https://github.com/ddnet/ddnet/pull/7489 was merged. This ensures tele still works and no new asan issues were added.

**Review by heinrich5991 — APPROVED** [review]
Thanks.

**Pull request merged**